### PR TITLE
checking bogon locally

### PR DESCRIPTION
**Pull request by rm-Umar**

### CODE REVIEW

**Comment by rm-Umar** [comment]
@UmanShahzad as `php 7.3` and `php 7.4` already reached end of life we'll only support `php 8.0>=`. Also updated the unit tests from `php 7.3, 7.4, 8.0` to `8.0, 8.1, 8.2`

**Comment by rm-Umar** [comment]
This will be a breaking change btw as the code will not work on `php8.0<` due to the module we are using for checking bogon locally so we need to release a `3.0` right after merging this.

File: src/IPinfo.php [hunk 1]
```diff
@@ -208,6 +209,13 @@ public function formatDetailsObject($details = [])
      */
     public function getRequestDetails(string $ip_address)
     {
+        if ($this->isBogon($ip_address)) {
+            return [
+                "ip" => $ip_address,
+                "bogon" => "True",
```
Comment (UmanShahzad) [inline src/IPinfo.php:215]: String? Is that how we currently represent bools when responding?

File: tests/IPinfoTest.php [hunk 2]
```diff
@@ -258,4 +258,32 @@ public function testGetBatchDetails()
             ]);
         }
     }
+
+    public function testBogonLocal4()
+    {
+        $tok = getenv('IPINFO_TOKEN');
```
Comment (UmanShahzad) [inline tests/IPinfoTest.php:264]: Are these really required for these bogon tests?

**Review by UmanShahzad — APPROVED** [review]

**Pull request merged**